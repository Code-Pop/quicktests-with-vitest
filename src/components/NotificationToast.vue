<template>
    <div role="alert" :class="[
        'notification',
        status === 'error' ? 'notification--error' : null,
        status === 'success' ? 'notification--success' : null,
        status === 'info' ? 'notification--info' : null,
        message && message.length > 0 ? 'notification--slide' : null,
    ]">
        <p class="notification__text">
            {{ message }}
        </p>
        <button data-testid="close-notification" title="close" ref="closeButton" class="notification__button"
            @click="$emit('clear-notification')">
            &#10005;
        </button>
    </div>
</template>

<script setup>
defineProps({
    status: {
        type: String,
        default: null
    },
    message: {
        type: String,
        default: null
    }
})
</script>

<style>
.notification {
    transition: all 900ms ease-out;
    opacity: 0;
    z-index: 300001;
    transform: translateY(-100vh);
    box-sizing: border-box;
    padding: 10px 15px;
    width: 100%;
    max-width: 730px;
    display: flex;
    position: fixed;
    top: 20px;
    right: 15px;
    justify-content: flex-start;
    align-items: center;
    border-radius: 8px;
    min-height: 48px;
    box-sizing: border-box;
    color: #fff;
}

.notification--slide {
    transform: translateY(0px);
    opacity: 1;
}

.notification__text {
    margin: 0;
    margin-left: 17px;
    margin-right: auto;
}

.notification--error {
    background-color: #fdecec;
}

.notification--error .notification__text {
    color: #f03d3e;
}

.notification--success {
    background-color: #e1f9f2;
}

.notification--success .notification__text {
    color: #146354;
}

.notification--info {
    background-color: #9AC7F5;
}

.notification__button {
    border: 0;
    background-color: transparent;
    cursor: pointer;
}
</style>