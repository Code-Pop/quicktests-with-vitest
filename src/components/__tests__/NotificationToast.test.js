import { mount } from '@vue/test-utils'
import NotificationToast from '../NotificationToast.vue'
import { describe, expect, test } from 'vitest'

describe('Notification component', () => {
  test('renders the correct style for error', () => {
    const status = 'error'
    const wrapper = mount(NotificationToast, {
      props: { status }
    })
    expect(wrapper.classes()).toEqual(expect.arrayContaining(['notification--error']))
  })

  test('renders correct style for success', () => {
    const status = 'success'
    const wrapper = mount(NotificationToast, {
      props: { status }
    })
    expect(wrapper.classes()).toEqual(expect.arrayContaining(['notification--success']))
  })

  test('renders correct style for info', () => {
    const status = 'info'
    const wrapper = mount(NotificationToast, {
      props: { status }
    })
    expect(wrapper.classes()).toEqual(expect.arrayContaining(['notification--info']))
  })

  test('slides up when message is empty', () => {
    const message = ''
    const wrapper = mount(NotificationToast, {
      props: { message }
    })
    expect(wrapper.classes('notification--slide')).toBe(false)
  })

  test('emits event when close button is clicked', async () => {
    const wrapper = mount(NotificationToast, {
      data() {
        return {
          clicked: false
        }
      }
    })
    const closeButton = wrapper.find('button')
    await closeButton.trigger('click')
    expect(wrapper.emitted())
    expect(wrapper.emitted()).toHaveProperty('clear-notification')
  })

  test('renders correct message to viewer', () => {
    const message = 'Something happened, try again'
    const wrapper = mount(NotificationToast, {
      props: { message }
    })
    expect(wrapper.find('p').text()).toBe(message)
  })
})
