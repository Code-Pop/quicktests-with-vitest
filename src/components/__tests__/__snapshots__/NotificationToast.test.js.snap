// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notification component > renders the correct style for error 1`] = `
"<div role=\\"alert\\" class=\\"notification notification--error\\">
  <p class=\\"notification__text\\"></p><button title=\\"close\\" class=\\"notification__button\\"> âœ• </button>
</div>"
`;
